qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -190,6 +191,22 @@ def __init__(self, canvas, num):
         self.window.setWindowTitle("Figure %d" % num)
         canvas.figure.set_label("Figure %d" % num)
 
+        # Set tight layout so axes labels remain visible
+        canvas.figure.set_layout_engine(layout="tight")
+        # Matplotlib issues a warnings if the window is made too small in the vertical or horizontal direction
+        # Filtering it to only warn once per plot per axis so that it doesn't give a warning everytime the plot window
+        # is resized, filling up the console with warnings
+        warnings.filterwarnings(
+            "once",
+            message="Tight layout not applied. The left and right margins cannot be made "
+            "large enough to accommodate all axes decorations.",
+        )
+        warnings.filterwarnings(
+            "once",
+            message="Tight layout not applied. The bottom and top margins cannot be made "
+            "large enough to accommodate all axes decorations.",
+        )
```

Comment:
I'm not sure I understand why there are two `filterwarnings` calls here?

Reply:
Those are two different warnings, one for the `left and right margins` and one for the `bottom and top margins`

Reply:
Ah right, I didn't spot the difference! In this case I'll approve